e2e/__tests__/typescriptConfigFile.test.ts

```diff
@@ -108,18 +112,20 @@ test('works with multiple typescript configs that import something', () => {
   expect(stdout).toBe('');
 });
 
-test("works with single typescript config that does not import anything with project's moduleResolution set to Node16", () => {
-  const {configs} = getConfig(
-    'typescript-config/modern-module-resolution',
-    [],
-    {
-      skipPkgJsonCheck: true,
-    },
-  );
+onNodeVersions('<=23.6', () => {
```

Comment:
Node resolution modes cannot be configured via `tsconfig.json` in modern node TS - it just uses modern node resolution by default.

This test fails anyways, because it requires transpilation to happen - the `package.json` contains `type: "commonjs"` and the `jest.config.ts` is written in ESM style. 
That fails native node. It would work with `type: "module"`, but we wouldn't really test what this test is about at this point - so I'm skipping it.